autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -508,6 +502,30 @@ def infer_objects(self, X: pd.DataFrame) -> pd.DataFrame:
 
         return X
 
+    def _is_datasets_consistent(self, diff_cols: List[Union[int, str]], X: pd.DataFrame) -> bool:
+        """
+        Check the consistency of dtypes between training and test datasets.
+        The dtypes can be different if the column belongs to `self.all_nan_columns`
+        (list of column names with all nans in training data) or if the column is
+        all nan as these columns would be imputed.
+
+        Args:
+            diff_cols (List[bool]):
+                The column labels that have different dtypes.
+            X (pd.DataFrame):
+                A validation or test dataset to be compared with the training dataset
+        Returns:
+            _ (bool): Whether the training and test datasets are consistent.
+        """
+        if self.all_nan_columns is None:
```

Comment:
when `self.all_nan_columns`, should you not check only for the second part at line 527?
it does not seem like it should return true, it seems like `inconsistent_cols` should be `diff_cols`.

Reply:
oh thats true, let me fix that

Reply:
I think this still does not fix it because it can be that `diff_cols` is not empty, but they are all `None`. So, the second `or `part of the return.

Reply:
```
 if self.all_nan_columns is None:
     if len(diff_cols) == 0:
         return True
     else:
         return False
```
If `self.nan_columns` is `None`, you check if there is a difference in the types by `len(diff_cols)`, if there is, you return `False`. Should you not also consider the case that there might be a difference in the types, however all of those columns might be NA fully for test?

`return len(inconsistent_cols) == 0 or all(X[inconsistent_cols].isna().all())` this is done by the second part of this or with 2 clauses. You are not reaching here and it might be that it is this case.

Maybe I am missunderstanding something?

Reply:
okay I got it, yeah it makes sense